core/src/main/java/org/kohsuke/stapler/RequestImpl.java

```diff
@@ -1158,6 +1166,7 @@ private boolean isMultipart() {
     }
 
     @Override
+    @WithBridgeMethods(value = org.apache.commons.fileupload.FileItem.class, adapterMethod = "fromFileUpload2FileItem")
```

Comment:
The Java compiler will generate a `getFileItem(String)` method that returns the new FileUpload 2.x type. The bridge method injector will call the abovementioned method, then call `fromFileUpload2FileItem` to generate the old type. Thus compatibility is retained.